src/cljs/org/broadinstitute/firecloud_ui/page/workspaces.cljs

```diff
@@ -211,12 +272,16 @@
      [:div {}
       (render-overlay state refs)
       [:div {:style {:padding "2em"}}
-       [:div {:style {:float "right" :display (when-not (:workspaces-loaded? @state) "none")}}
+       [:div {:style {:float "right" :display (when (or (not (:workspaces-loaded? @state))
+                                                        (:selected-workspace @state)) "none")}}
         [comps/Button
          {:text "Create New Workspace" :style :add
           :onClick #(swap! state assoc :overlay-shown? true)}]]
-       [:span {:style {:fontSize "180%"}} "Workspaces"]]
+       [:span {:style {:fontSize "180%"}} (if-let [ws (:selected-workspace @state)]
+                                            (ws "name")
+                                            "Workspaces")]]
       (cond
+        (:selected-workspace @state) (render-selected-workspace (:selected-workspace @state))
         (:workspaces-loaded? @state)
```

Comment:
I think the following `div` could be pulled out into a `render-workspaces-list` to make this `cond` a bit easier to follow.